Src/IronPython/Runtime/Types/PythonTypeInfo.cs

```diff
@@ -1728,7 +1728,8 @@ public LookupBinder(PythonBinder/*!*/ binder)
 
             // numeric types in python don't define equality, just __cmp__
             if (t == typeof(bool) ||
-                (Converter.IsNumeric(t) && t != typeof(Complex) && t != typeof(double) && t != typeof(float))) {
+                (Converter.IsNumeric(t) && t != typeof(Complex) && t != typeof(double) && t != typeof(float)) &&
+                t != typeof(Decimal)) {
```

Comment:
Do we eventually want this constraint to disappear altogether, if `__cmp__` is going away?

Reply:
Yeah, I think the whole method becomes obsolete once `__cmp__` is gone.